script.js

```diff
@@ -53,81 +79,247 @@ my_widget_script =
             }
             return true;
           }
-         
+
           $('#the_form tbody tr').each(function() {
-              if (isEmpty(this)) {
-                $(this).remove();
-              }
-            })
+            if (isEmpty(this)) {
+              $(this).remove();
+            }
+          })
           return;
         }
-       
+
         var nFixed = 2;
-        $('#the_form input[name^=amount]').on('keyup change', function() {
+
+        /*
+			Amount Field On Blur Handler
+		*/
+        $('#the_form input[name^=amount]').on('blur', function() {
           var tr = $(this).closest('tr');
-          var equiv = $('input[name^=equivalents]', tr);
-          if (equiv.length != 1)
-            throw ('Found the incorrect number of equiv inputs: ' + equiv.length)
-          equiv = equiv.first();
-          var moles1 = $('#the_form input[name=moles1_formula]');
+          console.log('In amount blur handler'+ tr.attr('class'))
+
+          current_class = tr.attr('class')
+          var compare_class_id = current_class.localeCompare('initialRow');
```

Comment:
`compare_class_id` is a very strange name for this:
* `compare` is a common verb. It should start the name of a function or method, but not a bool.
* this does not store an ID
* functionally the class is only a tool here: what you want the name to suggest is that it is a boolean indicator of whether the present context is the first row or a subsequent row.
* the name says nothing about being initial

Better names?
* `is_first_row`
* `is_initial_row`
* `initial`

Reply:
There should be no need to use `localCompare` for class names. Use `==`